### PR TITLE
Perf optimizations

### PR DESCRIPTION
**Pull request by divmain**
Originally https://github.com/FormidableLabs/victory-util/pull/27.

/cc @boygirl @coopy

### CODE REVIEW

**Comment by divmain** [comment]
Merging, and will address `_.memoize` if needed with a separate PR.

File: src/victory-util/perf.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Memoizes multi-argument functions.
+ *
+ * NOTE: If `fn` receives another function as its argument, memoization is
+ * not guaranteed to work.  The function will be strigified, but it will
```
Comment (ryan-roemer) [inline src/victory-util/perf.js:5]: `s/strigified/stringified/`

File: src/victory-util/perf.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * @param  {Function}  fn  The function to memoize
+ *
+ * @return {Function}      Memoized `fn`.
+ */
+export const memoize = function memoize(fn) {
+  const cache = {};
+  return function () {
+    const args = Array.prototype.slice.call(arguments);
+    const hash = args.map((arg) => {
```
Comment (ryan-roemer) [inline src/victory-util/perf.js:16]: Curious -- why not just `JSON.stringify(args);`?

File: src/victory-util/perf.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * NOTE: If `fn` receives another function as its argument, memoization is
+ * not guaranteed to work.  The function will be strigified, but it will
+ * no longer be a closure.
+ *
+ * @param  {Function}  fn  The function to memoize
+ *
+ * @return {Function}      Memoized `fn`.
+ */
+export const memoize = function memoize(fn) {
```
Comment (ryan-roemer) [inline src/victory-util/perf.js:12]: Also, tests would be nice and should be super easy to write. Let's make sure we cover our base and corner cases!

**Comment by ryan-roemer** [comment]
@divmain -- Got some discussion points, refactors and additions for a follow-on PR. Thanks!
